src/modular/safegcd/boxed.rs

```diff
@@ -33,572 +33,412 @@ impl BoxedSafeGcdInverter {
     /// Modulus must be odd. Returns `None` if it is not.
     pub fn new(modulus: &Odd<BoxedUint>, adjuster: &BoxedUint) -> Self {
         Self {
-            modulus: BoxedUnsatInt::from(&modulus.0),
-            adjuster: BoxedUnsatInt::from(&adjuster.resize(modulus.bits_precision())),
-            inverse: invert_mod_u64(modulus.0.as_words()) as i64,
+            modulus: modulus.clone(),
```

Comment:
If you're just cloning the modulus then we should probably just get rid of the borrow and have the caller clone if need be

Reply:
Updated now